src/lib/grad.jl

```diff
@@ -137,7 +137,19 @@ julia> gradient((a,t) -> sum(a .* t[1]) + t[2], [1,2,3], (4,5))  # gradient unde
 ([4 4 4], (6, 1))
 ```
 """
-function jacobian(f, args...)
+jacobian(f, args...) = withjacobian(f, args...).grad
+
+"""
+    withjacobian(f, args...)
+
+Returns both the value `f(args...)` and the [`jacobian`](@ref) as a named tuple.
+
+```jldoctest; setup=:(using Zygote)
+julia> withjacobian(cumsum, [1,2,3])
+(val = [1, 3, 6], grad = ([1 0 0; 1 1 0; 1 1 1],))
```

Comment:
Should it be `jacobian` instead of `grad`? Or we can standardize around `value` and `deriv`?

Reply:
I was hoping that `grad` was neutral enough, and abbreviated to a comparable extent to `val`. Zygote already has `Grads` for all purposes. 

Of course they are all derivatives, just with varying numbers of indices on... `deriv` somehow seems less pretty to me, and of course "derive" mostly means something unrelated to "derivative".

It could follow the function names exactly, but IMO it's a little weird to re-use a function name as a field name (when that function isn't the getter).

Reply:
How do you feel about `value` and `diff`?

Reply:
I would say `val` & `diff` as it seems odd to abbreviate only one. One objection is that `diff` is actually a function, but I doubt that'll confuse anyone. 

I could live with this, but still prefer `grad` I think. To my ear at least, it seems generic enough to mean "the thing AD produces" not specifically what calculus class called the gradient.

Reply:
Yeah perhaps you are right. I am okay with either `val`/`diff` or `val`/`grad` for this PR.

Reply:
Cool. Perhaps this is ready to go?

Reply:
sorry to come late to the party, but can't we bypass the naming issue entirely by returning a tuple? Returning a named tuple doesn't seem to add much additional value

Reply:
🤷🏾‍♂️ it's nice on the user end to make their code more readable. I don't think the current names are bad, I just wanted to make sure we discussed it.

Reply:
Yes I guess the downside is that naming things is hard, as they say. But IMO there are already too many tuples of tuples of iterators of tuples etc. floating around, hence lots of slightly mysterious `[2]`s, seems nice to have some optional signposts.

Not sure I'm brave enough see what happens if `pullback` returned a named tuple, though!